fix: sync running into loop on invalid keys

sitaram-kalluri opened the pull request:
**- What I did**
* Fix the sync infinite loop issue.

**- How I did it**
* At the moment, When the server is ahead of the local secondary, to ensure all the delta commits are received, we run a loop until the localCommitId is equal to serverCommitId.
* The problem with this approach is if an invalid key is not synced to local  secondary, the sync runs into an infinite loop.
* Hence we replaced the `while (serverCommitId > localCommitId)` with `while (serverCommitId > lastReceivedServerCommitId)` so that we run the loop until the lastReceivedServerCommitId is equal to serverCommitId

**- How to verify it**
* The end-to-end tests and the unit tests should pass.

**- Description for the changelog**
* Fix the sync running into an infinite loop when invalid keys do not sync to the local secondary

sitaram-kalluri committed:
fix: sync running into loop on invalid keys

gkc commented on packages/at_client/lib/src/service/sync_service_impl.dart:
```diff
@@ -422,7 +431,8 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
       }
```
LGTM. While we're changing the code here, can you also log the stack trace when we encounter exceptions or errors please

sitaram-kalluri replied:
Added stack trace to the exception and the error.

dependabot[bot] committed:
build(deps): bump peter-evans/create-pull-request from 4.2.1 to 4.2.2

Bumps [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) from 4.2.1 to 4.2.2.
- [Release notes](https://github.com/peter-evans/create-pull-request/releases)
- [Commits](https://github.com/peter-evans/create-pull-request/compare/d7db273d6c7206ba99224e659c982ae34a1025e3...331d02c7e2104af23ad5974d4d5cbc58a3e6dc77)

---
updated-dependencies:
- dependency-name: peter-evans/create-pull-request
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sitaram-kalluri committed:
fix: add stacktrace to exception

sitaram-kalluri committed:
fix: add stacktrace to exception

sitaram-kalluri committed:
Merge branch 'trunk' into fix_sync_infinite_loop
